proof/refine/AARCH64/ArchAcc_R.thy

```diff
@@ -426,36 +435,25 @@ lemma setObject_PT_corres:
     apply (drule_tac x = x in bspec)
      apply simp
     apply (rule conjI; clarsimp)
-     apply (clarsimp simp: pte_relation_def)
-    prefer 2
-  sorry (* FIXME AARCH64
-    apply (clarsimp simp: pte_relation_def table_base_index_eq
+     apply (clarsimp simp: pte_relation_def pt_apply_upd_eq')
+     apply (metis more_pt_inner_beauty)
+    apply (clarsimp simp: pte_relation_def table_base_index_eq pt_apply_upd_eq'
                    dest!: more_pt_inner_beauty)
-    apply (simp add: pt_apply_pt_upd_eq)
    apply (rule ballI)
    apply (drule (1) bspec)
    apply clarsimp
    apply (rule conjI)
-    apply (clarsimp simp: pte_relation_def table_base_index_eq
-                   dest!: more_pt_inner_beauty)
+    apply (clarsimp simp: pte_relation_def pt_apply_upd_eq')
+    apply (metis more_pt_inner_beauty table_base_index_eq)
    apply clarsimp
    apply (drule bspec, assumption)
    apply clarsimp
    apply (erule (1) obj_relation_cutsE)
       apply simp
-     apply simp
      apply clarsimp
-     apply (frule (1) pspace_alignedD)
-     apply (drule_tac p=x in pspace_alignedD, assumption)
-     apply simp
-     apply (drule mask_alignment_ugliness)
-        apply (simp add: pt_bits_def pageBits_def)
-       apply (simp add: pt_bits_def pageBits_def)
-      apply clarsimp
-      apply (drule test_bit_size)
-      apply (clarsimp simp: word_size bit_simps)
-      apply arith
-     apply ((simp split: if_split_asm)+)[2]
+     apply (smt (verit, best) pspace_aligned_pts_ofD pts_of_Some pts_of_type_unique aobjs_of_Some
+                              table_base_plus)
```

Comment:
`smt` is fine these days. The reason we didn't use it in the old days was that `smt` replay was either extremely brittle or not done. This is no longer the case -- the AFP also accepts it and we haven't seen any issues yet.

Despite that I also still have the reflex to prefer `metis` and co, but in this case those didn't work.

Reply:
I remain sceptical of `smt`, but can't substantiate it. Also, the indent's off for second line, since it came out of sledgehammer.

Reply:
Will fix the indent. `smt` is as solid as the other ones these days, but I do share the urge to avoid it. I think we've trained ourselves to avoid it for so long now that we can't stop without feeling like something is off..

Reply:
Actually, the indent looks fine to me. Where would you put it?

Reply:
It was showing up differently in discussion view. You're right, looks fine in file display.

Reply:
I also thought I saw it wrong initially. All good.